src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -137,66 +138,52 @@ def on_time_step(self, event: Event):
         move those households to a new address
         """
         households = self.population_view.get(event.index)
-        household_heads = households.loc[
+        household_ids = households.loc[
             households["relation_to_household_head"] == "Reference person"
-        ]
-        households_that_move = self.addresses.determine_if_moving(
-            household_heads["household_id"], self.household_move_rate
+        ]["household_id"]
+
+        all_household_ids_that_move = self.randomness.filter_for_rate(
+            household_ids,
+            self.household_move_rate(household_ids.index),
+            "all_moving_households",
         )
 
-        # Find households that move abroad and separate subsets of state table
-        households_that_move_abroad_idx = self.determine_households_moving_abroad(
-            households_that_move, self.proportion_households_leaving_country, event
+        # Determine which households move abroad
+        abroad_household_ids = filter_by_rate(
+            all_household_ids_that_move,
+            self.randomness,
+            self.proportion_households_leaving_country,
+            "abroad_households",
         )
-        moving_abroad_households = households_that_move.loc[households_that_move_abroad_idx]
-        moving_domestic_households = households_that_move.loc[
-            ~households_that_move.index.isin(moving_abroad_households.index)
-        ]
-        abroad_moving_households = households.loc[
-            households["household_id"].isin(moving_abroad_households)
-        ]
-        domestic_moving_households = households.loc[
-            households["household_id"].isin(moving_domestic_households)
-        ]
+        domestic_household_ids = ~all_household_ids_that_move.isin(abroad_household_ids)
```

Comment:
im surprise this works -- isn't this a series of bools?

Reply:
might need something like  `all_household_ids_that_move.loc[~all_household_ids_that_move.isin(abroad_household_ids)]`?